cmd/tracestat/main.go

```diff
@@ -103,28 +118,28 @@ func (ts *tracestat) load(f string) {
 	for {
 		evt.Reset()
 
-		err = pbr.ReadMsg(&evt)
-		if err != nil {
-			if err == io.EOF {
-				return
-			}
-
-			log.Printf("error decoding trace event from %s: %s", f, err)
-			return
+		switch err = pbr.ReadMsg(&evt); err {
+		case nil:
+			ts.addEvent(&evt)
+		case io.EOF:
+			return nil
+		default:
+			return fmt.Errorf("error decoding trace event from %s: %w", f, err)
 		}
-
-		ts.addEvent(&evt)
 	}
 }
 
 func (ts *tracestat) addEvent(evt *pb.TraceEvent) {
-	peer := peer.ID(evt.GetPeerID())
+	var (
+		peer      = peer.ID(evt.GetPeerID())
```

Comment:
I didn't touch this, but I think it's a bug. Shouldn't we be using `IDFromBase58` or `IDFromString`?

Reply:
No, it's a raw peer ID. No need to decode.